include/targets/generic/thread_array.h

```diff
@@ -22,20 +22,18 @@ namespace quda
   public:
     using Smem::shared_mem_size;
 
-    __device__ __host__ constexpr thread_array() : array_(sharedMem()[target::thread_idx_linear<3>()])
+    template <typename... U>
+    __device__ __host__ constexpr thread_array(const KernelOps<U...> &ops) :
+      Smem(ops), array_(sharedMem()[target::thread_idx_linear<3>()])
     {
-      array_ = array<T, n>(); // call default constructor
+      checkKernelOps<thread_array<T, n, O>>(ops);
+      array_ = array<T, n> {}; // call default constructor
     }
 
-    template <typename... Ts>
-    __device__ __host__ constexpr thread_array(T first, const Ts... other) :
-      array_(sharedMem()[target::thread_idx_linear<3>()])
-    {
-      array_ = array<T, n> {first, other...};
-    }
+    constexpr thread_array(const thread_array<T, n, O> &) = delete;
 
-    __device__ __host__ T &operator[](int i) { return array_[i]; }
-    __device__ __host__ const T &operator[](int i) const { return array_[i]; }
+    __device__ __host__ inline T &operator[](const int i) { return array_[i]; }
```

Comment:
No real reason, I've removed them and a few unnecessary `inline`s too (since member functions are inline by default).
3b71bb635343325aa92a9d5e0262303f1f4f7549